src/bootstrap/native.rs

```diff
@@ -789,6 +789,17 @@ impl Step for Lld {
             .define("LLVM_CONFIG_PATH", llvm_config_shim)
             .define("LLVM_INCLUDE_TESTS", "OFF");
 
+        if builder.config.llvm_thin_lto {
+            cfg.define("LLVM_ENABLE_LTO", "Thin");
+            if !target.contains("apple") {
+                cfg.define("LLVM_ENABLE_LLD", "ON");
```

Comment:
Why is apple special-cased here? why does LTO imply LLD?

Reply:
In Rust's build setup, it uses ThinLTO (`-flto=Thin`), this is a LLVM specific feature so require the usage of LLD.

Apple's Xcode does not ship with LLD, instead linking is done using its own linker `ld64`, which supports thin LTO out of the box (it kinda make sense as Apple's toolchain is based on LLVM).

Reply:
Ok. Can you add that as a comment please?

Reply:
Added comments as requested.